net_transport_test.go

```diff
@@ -76,8 +76,8 @@ func TestNetworkTransport_CloseStreams(t *testing.T) {
 		t.Fatalf("err: %v", err)
 	}
 	defer trans2.Close()
-
-	for i := 0; i < 2; i++ {
+	var i int
```

Comment:
Yeah we should definitely do a refactoring pass before we merge. This is  a bug waiting to happen IMHO